syntax = "proto3";

package user.v1;
import "errors/errors.proto";

option go_package = "api/user/service/v1/pb;v1";

enum UserErrorReason {
  option (errors.default_code) = 500;

  UNKNOWN_ERROR = 0;
  VERIFY_PASSWORD_FAILED = 1 [(errors.code) = 401];
  VERIFY_CODE_FAILED = 2 [(errors.code) = 401];
  USERNAME_CONFLICT = 3 [(errors.code) = 409];
  PHONE_CONFLICT = 4 [(errors.code) = 409];
  EMAIL_CONFLICT = 5 [(errors.code) = 409];
  WECHAT_CONFLICT = 6 [(errors.code) = 409];
  QQ_CONFLICT = 7 [(errors.code) = 409];
  GITEE_CONFLICT = 8 [(errors.code) = 409];
  GITHUB_CONFLICT = 9 [(errors.code) = 409];
  USER_NAME_CONFLICT = 10 [(errors.code) = 409];
  GET_ACCOUNT_FAILED = 11 [(errors.code) = 404];
  GET_PROFILE_FAILED = 12 [(errors.code) = 404];
  GET_PROFILE_UPDATE_FAILED = 13 [(errors.code) = 404];
  GET_ACHIEVEMENT_FAILED = 14 [(errors.code) = 404];
  GET_USER_FOLLOW_FAILED = 15;
  GET_FOLLOW_LIST_FAILED = 16;
  GET_FOLLOW_LIST_COUNT_FAILED = 17;
  GET_PROFILE_LIST_FAILED = 18;
  GET_USER_SEARCH_FAILED = 19;
  GET_PICTURE_REVIEW_FAILED = 20;
  SEND_CODE_FAILED = 21;
  SET_PHONE_FAILED = 22;
  SET_EMAIL_FAILED = 23;
  SET_PROFILE_FAILED = 24;
  SET_PROFILE_UPDATE_FAILED = 25;
  SET_USERNAME_FAILED = 26;
  SET_PASSWORD_FAILED = 27;
  SET_FOLLOW_FAILED = 28;
  SET_IMAGE_FAILED = 29;
  SET_IMAGE_IRREGULAR_FAILED = 30;
  SET_WECHAT_FAILED = 31;
  SET_QQ_FAILED = 32;
  SET_GITEE_FAILED = 33;
  SET_GITHUB_FAILED = 34;
  CANCEL_FOLLOW_FAILED = 35;
  LOGIN_FAILED = 36;
  REGISTER_FAILED = 37;
  RESET_PASSWORD_FAILED = 38;
  PROFILE_REVIEW_MODIFY_FAILED = 39;
  PROFILE_UPDATE_MODIFY_FAILED = 40;
  UNBIND_ACCOUNT_FAILED = 41;
  UNIQUE_ACCOUNT = 42;
}
